File: 2582-pass-the-pillow/2582-pass-the-pillow.cpp
class Solution {
public:
    int passThePillow(int n, int time) {
        int curr = 1, sign = 1;
        
        while (time-- > 0){
            curr += sign;
            
            if (curr == 1 || curr == n)
                sign *= -1;
            
        }
        
        return curr;
    }
};

File: 2582-pass-the-pillow/2582-pass-the-pillow.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        
        curr, sign = 1, 1
        while time > 0:
            # Pass the pillow
            curr += sign
            
            if curr == 1 or curr == n:
                sign *= -1
            
            time -= 1
            
        return curr